File: ts/src/resources/health.ts
import { HttpTransport } from "../http.js"

export interface HealthStatus {
  status: string
  uptime?: number
  version?: string
}

export class HealthResource {
  constructor(private readonly http: HttpTransport, private readonly base: string) {}

  async check(): Promise<HealthStatus> {
    return this.http.getFetch()(`${this.base}/`, { method: "GET" })
      .then(res => res.json())
  }
}

File: py/src/imbrace/app/resources/health.py
from typing import Any
from ...core.http import HttpTransport, AsyncHttpTransport


class HealthResource:
    def __init__(self, http: HttpTransport, base: str):
        self._http = http
        self._base = base

    def check(self) -> Any:
        return self._http.request("GET", f"{self._base}/global/health").json()


class AsyncHealthResource:
    def __init__(self, http: AsyncHttpTransport, base: str):
        self._http = http
        self._base = base

    async def check(self) -> Any:
        res = await self._http.request("GET", f"{self._base}/global/health")
        return res.json()
